src/main/kotlin/com/petqua/presentation/auth/AuthExtractor.kt

```diff
@@ -0,0 +1,58 @@
+package com.petqua.presentation.auth
+
+import com.petqua.common.util.getCookieValueOrThrow
+import com.petqua.common.util.getHeaderOrThrow
+import com.petqua.common.util.throwExceptionWhen
+import com.petqua.domain.auth.token.AccessTokenClaims
+import com.petqua.domain.auth.token.JwtProvider
+import com.petqua.exception.auth.AuthException
+import com.petqua.exception.auth.AuthExceptionType.EXPIRED_ACCESS_TOKEN
+import com.petqua.exception.auth.AuthExceptionType.INVALID_ACCESS_TOKEN
+import com.petqua.exception.auth.AuthExceptionType.INVALID_AUTH_COOKIE
+import com.petqua.exception.auth.AuthExceptionType.INVALID_AUTH_HEADER
+import io.jsonwebtoken.ExpiredJwtException
+import io.jsonwebtoken.JwtException
+import jakarta.servlet.http.HttpServletRequest
+import org.springframework.http.HttpHeaders.AUTHORIZATION
+import org.springframework.stereotype.Component
+
+private const val AUTHORIZATION_PREFIX = "Bearer "
+private const val EMPTY = ""
+private const val REFRESH_TOKEN = "refresh-token"
+
+@Component
+class AuthExtractor(
+    private val jwtProvider: JwtProvider,
+) {
+
+    fun extractHeaderToken(request: HttpServletRequest): String {
+        val header = request.getHeaderOrThrow(AUTHORIZATION) { AuthException(INVALID_AUTH_HEADER) }
+        return parse(header)
+    }
+
+    private fun parse(header: String): String {
+        validateHeader(header)
+        return header.replace(AUTHORIZATION_PREFIX, EMPTY)
```

Comment:
```suggestion
        return header.removePrefix(AUTHORIZATION_PREFIX)
```
kotlin에서 좀 더 직관적인 함수도 제공하고 있습니다!

Reply:
대체 그런거 언제 다 배운거에요...